`Programming exercises`: Rename Java project types

ole-ve opened the pull request:
### Checklist
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [X] I documented the Java code using JavaDoc style.
#### Client
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [X] I added multiple screenshots/screencasts of my UI changes.
- [X] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
The names of the project types are ambiguous because they only refer to the project type of the template and solution repository and Maven is implicitly the type for the test repository. When [introducing Gradle as a new project type](https://github.com/ls1intum/Artemis/pull/4781), this has to be changed because there also has to be the option for plain template and solution repository with a Gradle test repository. This would interfere with the current project type 'Eclipse'.
In addition, the name 'Eclipse' is not completly correct because the project can also be opened in other IDEs but instead are basically only plain Java.

### Description
Within these changes, the project type 'Eclipse' has been renamed to 'Plain_Maven' and 'Maven' to 'Maven_Maven'. Therefore, the occurrences of these types have been changed as well as the names of the folders where the template exercise repository files are stored in.
However, no database change is required because the DB only saves the index of project type in ProjectType.java, but not an actual name. Since the order has not been changed, this does not cause any problem.
For Java exercises, the form for creating an exercise has separate forms for the project types of template/solution repository and the test repository.
As discussed with @krusche, these changes already include the additional enum values for the Gradle project types to be introduced. The reason behind this is to prevent errors when deploying other PRs after the PR adding Gradle support has been deployed. Those would occur since exercises from this new type and the corresponding enum values would not exist in other deployed PRs.

### Steps for Testing
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises
4. Confirm that old Java exercises are still working.
5. Generate Java programming exercises for the following configurations:
- Project Type: 'Plain Java' and Test Repository Project Type 'Maven'
- Project Type: 'Maven' and Test Repository Project Type 'Maven'.

Play around with the possible configurations (such as package name, SCA, and sequential test runs). Confirm that everything just works as expected.

6. Import a Java programming exercise for both project types and confirm that the repositories and build plans are set up correctly. Check this for both an exercise that has been created before and after this PR has been deployed.

### Review Progress
#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
\- No change - 

### Screenshots
<img width="890" alt="Bildschirmfoto 2022-03-08 um 15 22 28" src="https://user-images.githubusercontent.com/73833780/157256715-ac0f893a-662d-4099-993e-fb91e2ef51b7.png">

julian-christl commented:
I'm not sure about this change.
Wouldn't it make more sense to actually offer two settings? The tests in `Gradle` and `Maven` and Solution and Template in `Plain Java`, `Maven` and `Gradle`?
This is definitely way more work to do as you probably have to adapt more code on the way but it feels like this is more the cheap way (no offence please :D) but it's not very intuitive that you change two things at a time with one dropdown.

ole-ve commented:
> I'm not sure about this change. Wouldn't it make more sense to actually offer two settings? The tests in `Gradle` and `Maven` and Solution and Template in `Plain Java`, `Maven` and `Gradle`? This is definitely way more work to do as you probably have to adapt more code on the way but it feels like this is more the cheap way (no offence please :D) but it's not very intuitive that you change two things at a time with one dropdown.

I fully understand your suggestion. In fact, my initial approach was the following:
- Two Project Types (Maven and Gradle)
- Boolean value if the template & solution repository projects are plain (false) or should also be of the corresponding type (true)
In my opinion, there is no scenario where the combination Maven (Template & Solution) - Gradle (Test) and vice versa would make sense.

After discussing this idea with Stephan, he highly suggested me to take the "cheap" (^^) solution. My approach (and any other approach splitting up the type) would add more DB columns and has been considered too complex.
As a compromise, I would suggest solving this issue on the client-side. However, this increases the complexity of template code while only being needed for one language and there is a tooltip providing a description of the type.

Hialus commented:
This is a sensible refactoring and the changes look good.
But I agree with Julian that 2 dropdowns would be better.
This could be implemented as a purely client-side functionality. The project types can stay as they are now. So I think this would be feasible in the scope of this PR.

I am also not too sure about the "Plain" descriptor. It is not directly clear what "Plain" means in this context. I think at least a short description should be included.

julian-christl commented:
I could get behind the idea to keep the enum in the backend => no extensive changes necessary and add the functionality to the client => still good UX.

ole-ve commented:
> I could get behind the idea to keep the enum in the backend => no extensive changes necessary and add the functionality to the client => still good UX.

I have implemented the changes. Since the new project type (Gradle) does not exist right now, I think it's quite hard to verify that this works correctly. In case you want to check this, you can clone and modify it locally. Otherwise, I have already implemented it locally in a stash and could show you. Just hit me up.

julian-christl commented on src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts:
```diff
@@ -261,14 +276,39 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
+     * Updates the test repository project type. This only applies to Java exercises with project type 'Plain Java'
+     * @param type the type to update to
+     */
+    set selectedTestRepositoryProjectType(type: JavaTestRepositoryProjectType) {
```
I don't think this makes sense as you currently only display the maven option if a maven option is selected above.

ole-ve replied:
Removed in a6da54c.

julian-christl reviewed (dismissed):
Code looks good!

julian-christl reviewed (approved):
Reapprove code
